### DISCUSSION TITLE
Time to GPS fix? / mounting orientation of BN-880 towards sky?

### DISCUSSION THREAD

**Discussion by wanderschikanentv** [renatobo/bonogps#38] [Q&A] [answered]
Hi all,

I just created my build of the bonoGPS, and it works fine, except the fact that the time to GPS fix is very long, in the order of 20-25 minutes. Is this normal, or is something wrong with my build?

I power the bonoGPS via USB cable. After plugging the power in, the blue LED flashes 2-3 times slowly, and afterwards it starts blinking very quickly until the GPS fix is highlighted via blinking of red+blue LED.

I'm using a BN880 together with this ESP32: https://www.az-delivery.de/en/products/esp-32-dev-kit-c-v4

By the way, which side of the BN-880 should be pointed towards the sky? The big square patch side, or the side with the connector? Or doesn't matter?

With best regards,
Michael

**Comment by renatobo** [comment]
Hi Michael
20-25 minutes is a very long time. 

Tips:

- The antenna needs to be pointed to the sky: it's the ceramic large patch and it should have a CIROCOMM writing on it
- The connector is on the bottom of the unit (as well as the BN880 writing)
- If you are testing indoors, you want to have the antenna looking outside a window and as close as possible to it

After the first time, a new boot should be much quicker (< 1 min) for the following few hours after the unit is turned off.

**Comment by wanderschikanentv** [comment]
Thanks for your answer - very clear about the orientation now! So, since you not mentioning my comment about the blinking pattern of the LEDs, I'd assume, it is probably no issue about e.g. quality of wiring/soldering, right?

**Comment by renatobo** [comment]
As long as you power the GPS unit, its lights are independent of what happens on the ESP32 side. Have you connected your unit to a mobile app to see if data shows up?

**Reply by wanderschikanentv** [reply]
Yes, I tried Racechrono and Harry's Laptimer, both are reading proper 10 Hz data after fixing. I think my issue is related to how I mounted the BN880. I sandwiched it with double-sided tape on the ESP32 to get a compact design. However, first time, I had the antenna oriented towards the ESP32. Now, I turned the antenna away from the ESP32, and it seems to work better. Still, I'm wondering whether I should give up that sandwich design and move the BN880 away from the ESP, maybe even in a separate housing. How is your experience on that topic? Will the BN880 be disturbed by the ESP being very close?

**Comment by renatobo** [comment]
I would recommend keeping the ESP32, at least its antenna, as far as reasonably possible from the GPS antenna. You should be fine without a separate housing, the best location to reduce interference is right below - you can read more about interference and how the GPS antenna works at (https://www.u-blox.com/sites/default/files/products/documents/GNSS-Antennas_AppNote_%28UBX-15030289%29.pdf)

**Reply by wanderschikanentv** [reply]
I'm pretty sure now that my issue was related to mounting the antenna incorrectly upside-down. Now, after changing, even in the sandwich on the ESP, the bonogps works like a charm :-)

One thing on the configuration: Currently, I'm using "Stream GxGBS" only  like on your screenshots and I'm getting the full 10 Hz. GxGBS seems to be sufficient for HLT and Racechrono. But what is behind "Stream GxGVA" and "Stream GxGSA" options? I tried them, but with all 3 options activated, update rate dropped below 10 Hz, say 7-9 Hz. I saw that HLT would show the position of the GPS satellites explicitely which is nice to know, but not mandatory, but on top of that, I didn't perceive additional benefit for the actual tracking.

Can you shed some light on which of these options you are recommending?

**Comment by renatobo** [comment, marked as answer]
Hint: there are predefined configurations for HLT and RaceChrono that document the optimal settings for each, see also documentation at (https://github.com/renatobo/bonogps/tree/main/software/connecting/harrylaptimer#gps-configuration-for-nmea-messages) and (https://github.com/renatobo/bonogps/tree/main/software/connecting/racechrono#gps-configuration-for-nmea-messages)

Some context

- GBS is optional to improve the quality of data, not all apps use it. RaceChrono for example does not (see footnote on the RaceChrono page listed above). A GBS message is rather small
- GSA and GSV provide information on the position of satellites in view: totally unnecessary to perform the timing functions, very useful to understand what your GPS unit is currently receiving. GSV and GSA are very large messages, 3 or 4 times the data required by position and time.

The "stream" option means that every command is sent at every refresh by the GPS unit, so 1Hz/5Hz/10Hz
The "poll" option means that the ESP32 device periodically requests commands at a different frequency

**When you activate stream for GSV/GSA, there is not enough bandwidth at anything higher than 1Hz (issues on both the GPS->ESP32 serial port AND on the BLE/BT connection) so refresh frequency falls back.**

Bottom line: Streaming GSV/GSA is not recommended but for diagnosing purposes. Use GSA and GSV when tuning your device, then you can disable it altogether for optimal performances. If you like the "eye candy" feature of checking how many satellites you see and where they are, a good option is "poll" which asks for that information much less frequently.

Read also [on the HLT forum](http://forum.gps-laptimer.de/viewtopic.php?t=4359)
